src/pyhf/pdf.py

```diff
@@ -188,7 +188,7 @@ def _nominal_and_modifiers_from_spec(config, spec):
                     )  # broadcasting
                 elif mtype in ['shapesys', 'staterror']:
                     uncrt = thismod['data'] if thismod else [0.0] * len(nom)
-                    maskval = [True if thismod else False] * len(nom)
+                    maskval = [(x > 0 and y > 0) for x, y in zip(uncrt, nom)]
```

Comment:
this is affecting `staterror` too. Is this intended?

Reply:
the logic should be the same but I haven't looked at staterror, prob adding the mast doesn't hurt

Reply:
staterror will only hit this if the entire bin is zero (i.e. none of the samples contribute) which is probably rare